### ISSUE TITLE
Discussion and a few observations

### ISSUE THREAD

**Issue by ecowitness** [UKPLab/EasyNMT#22]
First -- thank you for the docker container. I could not get the Dockerfile to build. I am impressed with my results so far. 

Second -- during startup, I get the following :
```

Booted as backend: True
Load model: opus-mt
[2021-03-30 00:17:10 +0000] [2255] [INFO] Started server process [2255]
[2021-03-30 00:17:10 +0000] [2255] [INFO] Waiting for application startup.
[2021-03-30 00:17:10 +0000] [2255] [INFO] Application startup complete.
/usr/local/lib/python3.8/site-packages/transformers/tokenization_utils_base.py:3216: FutureWarning: `prepare_seq2seq_batch` is deprecated and will be removed in version 5 of 🤗 Transformers. Use the regular `__call__` method to prepare your inputs and the tokenizer under the `with_target_tokenizer` context manager to prepare your targets. See the documentation of your specific tokenizer for more details
  warnings.warn(
```

Third -- some translations are humerous (using opus-mt). consider sv to en (Swedish to English) of sitt, which translates to:

This medicine is not recommended for use in patients with severe hepatic impairment.

or stackers, which translates to 

poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor, poor.

(You can confirm these by using the $10 computer translation window and swedish to english).

Translation of än goes on for quite a few lines. Or you can do vågar

Fourth -- I'm running on some pretty good iron and the performance seems a little slow. Are there some tuning issues that I can address? I have enough cpu and memory.

Fifth -- A sincere thanks for putting this together. I was using googletrans and it was giving me fits.

And you might care about what I'm using easyNMT for....I'm doing prep work for a student taking classes in a foreign language. I am pulling the English closed captions out of the recording of a broadcast TV program, translating them to Swedish, and then putting them back into the program before we watch it. I also build a new word list from the Swedish translations, eliminate words the student already knows, and create a flat file of the Swedish word and English meaning that can be input to a flash card program.  This is working much better since I have started using easyNMT. Many, many thanks.

**Comment by nreimers** [comment]
Hi,
1) Is there any error?

2) This can be ignored

3) Yes, the opus-mt models has some issues. It was trained on a rather small dataset consisting of sentences. If you translated words or short phrases with it, it can lead to these strange behavior. The M2M models are more robust (im my impression). Sadly there is not easy fix for the opus-mt models

4) Running on a GPU significantly improves the performance. But otherwise, there is sadly not much tuning possible as transformers are quite compute expensive on CPU. You can use Google Colab to get a free GPU. See the links in the Readme for some examples how to run the code or a REST API on Google Colab.

**Comment by ecowitness** [comment]
regarding:
1. I originally started out trying to build my own image using pip3 to install easynmt. Build failed a number of times. I would try to figure out what would satisfy the issue and pip3 install that before the pip3 install easynmt so the requirement would be met. I ran out of the allocated time before I got it working. The last issue that got me red text was sentencepiece. Next day I pulled your image and have been using it since.
2. Consider it done.
3. I tried to load m2m_100_1.2B but it went into a loop tripping a watchdog timer. Then I loaded m2m_100_418M, but it translated "eighteen" to "18" and that isn't much use when building word match flashcards. I went back to opus-mt. We are doing a few manual corrections, as needed. Most of the translating that we are doing is for one time use. Most TV isn't worth watching a second time.
4. I understand the need for a gpu. I'm running on a Dell PowerEdge R610 and there really isn't an easy way to get a gpu on board. I translated the Swedish CEFR Kelly list yesterday (8500 words, one at a time) and that went quickly enough. When we are doing a TV program, there are about 2500 lines of closed captions in a 90 minute program and so far that has been about 1700 unique words. Some are, of course, people and place names. I am getting good enough performance to just keep chugging along.
5. Again.

Feel free to close this issue when you tire of discussion.
